src/backend/joanie/payment/backends/lyra/__init__.py

```diff
@@ -0,0 +1,135 @@
+            "amount": int(order.total * 100),
+            "currency": settings.DEFAULT_CURRENCY,
+            "customer": {
+                "email": order.owner.email,
+                "billingDetails": {
+                    "firstName": billing_address["first_name"],
+                    "lastName": billing_address["last_name"],
+                    "address": billing_address["address"],
+                    "zipCode": billing_address["postcode"],
+                    "city": billing_address["city"],
+                    "country": billing_address["country"],
+                    "language": order.owner.language,
+                },
+                "shippingDetails": {
+                    "shippingMethod": "DIGITAL_GOOD",
+                },
+            },
+            "orderId": str(order.id),
+            "ipnTargetUrl": self.get_notification_url(request),
+        }
+
+    def _call_api(self, url, payload, order_id):
+        """
+        Call the Lyra API with the given payload.
+
+        https://docs.lyra.com/fr/rest/V4.0/api/kb/webservice_usage.html
+        """
+        try:
+            response = requests.post(url, json=payload, headers=self.headers, timeout=5)
+            response.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            logger.error(
+                "Error calling Lyra API | %s: %s",
+                e.__class__.__name__,
+                e,
+                extra={
+                    "context": {
+                        "order_id": order_id,
+                        "url": url,
+                    }
+                },
+            )
+            return None
+
+        response_json = response.json()
+
+        if response_json.get("status") == "ERROR":
+            logger.error(
+                "Error calling Lyra API %s | %s: %s",
+                url,
+                response_json.get("answer").get("errorCode"),
+                response_json.get("answer").get("errorMessage"),
+                extra={
+                    "context": {
+                        "order_id": order_id,
+                        "url": url,
+                        "response_json": response_json,
+                    }
+                },
+            )
+            return None
+
+        if settings.DEBUG:
+            with open("response.json", "w", encoding="utf-8") as f:
+                json.dump(response_json, f, indent=4)
+
+        return response_json
+
+    def create_payment(self, request, order, billing_address):
+        """
+        Create a payment object for a given order
+        """
+        payment_data = self._get_payment_data(request, order, billing_address)
+        payment_data["formAction"] = "ASK_REGISTER_PAY"
+
+        url = f"{self.api_url}Charge/CreatePayment"
+        response_json = self._call_api(url, payment_data, order.id)
+
+        if not response_json:
+            return None
+
+        return response_json.get("answer").get("formToken")
```

Comment:
Just to make the code more robust, WDYT?
```suggestion
        return response_json.get("answer", {}).get("formToken")
```

Reply:
Even if there is the key `answer` in the response, will `formToken` always be available ?

What should be the default fallback value in the case where `formToken` is not available ?

Reply:
Answer key is always here, even when an error response comes from lyra.

I don't get what you mean by `fallback value`, we return the token or None.

Reply:
I meants will it return None, or will it return for example an empty dictionary.